ravendb/resource_ravendb_server.go

```diff
@@ -0,0 +1,385 @@
+						return warns, errs
+					} else if response.StatusCode != http.StatusOK {
+						errs = append(errs, fmt.Errorf("'%s' is not reachable. HTTP status code: %d. Please check the input version: %s. Url used was: %s", link, response.StatusCode, v, link))
+						return warns, errs
+					}
+					return warns, errs
+				},
+			},
+			"insecure": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whatever to allow to run RavenDB in unsecured mode. This is ***NOT*** recommended!",
+			},
+			"url": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"list": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+							},
+							MinItems: 1,
+						},
+						"http_port": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IsPortNumber,
+						},
+						"tcp_port": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IsPortNumber,
+						},
+					},
+				},
+			},
+			"settings": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"files": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsBase64,
+				},
+			},
+			"ssh": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"user": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"pem_base64": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsBase64,
+						},
+					},
+				},
+			},
+			"nodes": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The state of all the nodes in the RavenDB cluster.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"license": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"settings": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"certificate_base64": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"http_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tcp_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"files": {
```

Comment:
what's this for?

Reply:
It will go to `etc` by default right? Can we make it so they need to pass absolute paths so it's more explicit?

E.g.
```
  assets = {
    "/etc/ravendb/cert-load.sh" = filebase64("./scripts/cert-load.sh"), 
  }
```

Also need to `chown ravendb:ravendb $FILEPATH`